jetty-deploy/src/main/java/org/eclipse/jetty/deploy/DeploymentManager.java

```diff
@@ -237,6 +238,14 @@ protected void doStart() throws Exception
         {
             startAppProvider(provider);
         }
+
```

Comment:
If you used MultiException, you could lazily create it in addOnStartupError, and then add the throwable to it directly.

Reply:
I could, and I'm considering it.
I like it's `ifExceptionThrow()` which would throw the specific exception if that's the only one tracked.